tasks/parameter/sp.yml

```diff
@@ -1,28 +1,29 @@
+---
 - name: Load service_principal in ansible auth parameter
   set_fact:
-      service_principal: "{{ client_id }}"
-      client_secret: "{{ secret }}"
+    service_principal: "{{ client_id }}"
+    client_secret: "{{ secret }}"
   when: service_principal is not defined
-  ignore_errors: yes
+  ignore_errors: true
 
 - name: Load service_princial in environment variables
   set_fact:
-      service_principal: "{{ lookup('env', 'AZURE_CLIENT_ID') }}"
-      client_secret: "{{ lookup('env', 'AZURE_SECRET') }}"
+    service_principal: "{{ lookup('env', 'AZURE_CLIENT_ID') }}"
+    client_secret: "{{ lookup('env', 'AZURE_SECRET') }}"
   when: service_principal is not defined
-  ignore_errors: yes
+  ignore_errors: true
 
 - name: Load service_princial in ~/.azure/credentials
   set_fact:
-      service_principal: "{{ lookup('ini', 'client_id section=default file=~/.azure/credentials') }}"
-      client_secret: "{{ lookup('ini', 'secret section=default file=~/.azure/credentials') }}"
-  when: service_principal is not defined or service_principal == ""
-  ignore_errors: yes
+    service_principal: "{{ lookup('ini', 'client_id section=default file=~/.azure/credentials') }}"
+    client_secret: "{{ lookup('ini', 'secret section=default file=~/.azure/credentials') }}"
+  when: (service_principal is undefined) or (service_principal is defined and not service_principal | trim | length > 0)
```

Comment:
Hello, I am just curious about the benefit of this change.

Reply:
No benefit regarding code. Just following best practices (See ansiblelint at https://github.com/ansible/ansible-lint)